src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java

```diff
@@ -733,7 +760,7 @@ private boolean checkForRatedAndAssessedResult(Result result) {
      * @return true if the result is manual and the assessment is over, or it is an automatic result, false otherwise
      */
     private boolean checkForAssessedResult(Result result) {
-        return result.getCompletionDate() != null && ((result.isManual() && ExerciseDateService.isAfterAssessmentDueDate(this)) || result.isAutomatic());
+        return result.getCompletionDate() != null && ((result.isManual() && ExerciseDateService.isAfterAssessmentDueDate(this)) || result.isAutomatic() || result.isAutomaticAI());
```

Comment:
Would it be possible to name „isAutomaticAI“ a bit more concise? Or just „isAI“?